### ISSUE TITLE
VSCode refuses to stop at breakpoints

### ISSUE THREAD

**Issue by dannypike** [microsoft/vscode#82142]
Issue Type: <b>Bug</b>

I have tried so many combinations of webRoot, sourceMaps and sourceMapPathOverrides entries in launch.json while trying to get VSCode to stop at a breakpoint in the renderer process for my webpack-Electron-React app that I am beginning to despair.

Please make VSCode tell me what it cannot find and where it is looking for the thing that it cannot find. ChromeDevTools is quite happy to stop at the breakpoint in the webpack-ed file and VSCode is quite happy to stop at the main process code, but it ignores the renderer process that I am debugging at the same time.

I'm not asking for a lot of examples of how it works for other people (I've tried editing lots of those, too). I am asking for VSCode to tell me why it isn't able to work in my case, i.e. some useful debug output on what it has found and where it is looking for stuff.

Many thanks

VS Code version: Code 1.38.1 (b37e54c98e1a74ba89e03073e5a3761284e3ffb0, 2019-09-11T13:35:15.005Z)
OS version: Windows_NT x64 10.0.17763

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700T CPU @ 2.80GHz (8 x 2808)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.95GB (14.45GB free)|
|Process Argv|D:\work\gml\serenity\0.2\src\samples\ts\1_shapes|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (11)</summary>

Extension|Author (truncated)|Version
---|---|---
code-gnu-global|aus|0.2.2
npm-intellisense|chr|1.3.0
vscode-npm-script|eg2|0.3.9
vscode-mocha-test-adapter|hbe|2.1.2
vscode-test-explorer|hbe|2.14.6
vs-color-picker|lih|1.0.0
cpptools|ms-|0.25.1
debugger-for-chrome|msj|4.12.0
vscode-react-native|msj|0.12.0
psi-header|psi|1.8.0
markdown-preview-enhanced|shd|0.4.3

</details>

**Comment by roblourens** [comment]
I recommend starting with this to debug it: https://github.com/microsoft/vscode-chrome-debug#the-scripts-command

**Comment by dannypike** [comment]
Thank you; that's an extremely helpful command!

May I change the request to ask that you add this information to a tool-tip that appears when you hover over a break-point that refuses to activate? Knowing about that command would have saved me a lot of aggravation this morning.

**Comment by roblourens** [comment]
Well, there are a lot of reasons that a breakpoint will not be bound, they can come up in normal usage, and I don't want to cram all the docs into the tooltip. Sounds like you got it working?

**Comment by dannypike** [comment]
I did, thank you.

You shouldn't need to include all of the docs, just a simple hint that reduces the work that your users need to do to fix the problem. It's all about cost-benefit and the law of Diminishing Returns. In C++, I get a useful tool-tip like "the source code has changed", "there is no code for this line" when it does not bind. With a tip like that, I get some useful information about how to fix it, whereas a tip like "Unverified breakpoint" doesn't really tell me much. I can already see that a VSCode has not bound because the icon has changed from solid-red to hollow-grey.

Now that I know this command exists, I can see that people (you included) have mentioned it in some of the web matches for "Unverified breakpoints". But those comments are buried a long way down each set of comments and were easily missed by this user. I'm not asking you to anticipate every possibility, just react to the ones that you do see being encountered by your users, and fix those.

If you change the line that says something like this:
`
if (doesNotBind) {
  setTooltip("Unverified breakpoint")
}
`
to something like this:
`
if (doesNotbindBind) {
  setTooltip("Script failed to load; use .scripts command")
}
`

You would help users like me a lot, as I would then be searching for "VSCode .scripts command" and get more targetted and useful info than I got when I searched for "VSCode Unverified breakpoint".
